src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -3930,11 +3930,15 @@ public static String join(final boolean[] array, final char delimiter, final int
         if (endIndex - startIndex <= 0) {
             return EMPTY;
         }
-        final StringJoiner joiner = newStringJoiner(delimiter);
+        final StringBuilder stringBuilder = new StringBuilder();
         for (int i = startIndex; i < endIndex; i++) {
-            joiner.add(String.valueOf(array[i]));
+            stringBuilder
+                    .append(array[i])
+                    .append(delimiter);
         }
-        return joiner.toString();
+        return stringBuilder
+                .deleteCharAt(stringBuilder.length() - 1)
+                .toString();
```

Comment:
Interesting, I thought instead of `deleteChartAt` using an `if` statement in the `for` above, to avoid adding the last `delimiter` would improve more the performance. For a follow-up discussion I think.